### PR TITLE
Added gradients + tests for ImageProjectiveTransform

### LINKED ISSUE #9423: Feature Request : Gradients for tf.contrib.image.rotate()
I wish to propagate gradients through image rotation. This is currently not supported:

```
import tensorflow as tf
import numpy as np

images = tf.zeros(shape=[5, 10, 10, 1])
angles = 2 * np.pi * np.random.random(5) - np.pi
out = tf.contrib.image.rotate(images, angles)
out_sum = tf.reduce_sum(out)
images_grad = tf.gradients(out_sum, [images])

print images_grad
```

gives

```[None]```

### PR DESCRIPTION
**Pull request by sebastian-schlecht**
Fixes #9423 by providing gradients for the base op.

Gradients are computed using the inverse of the transformation matrix which can be slow (even though it's 3x3 matrices only) but allows for dynamic transformation matrices. In case the transform is static, this might be precomputed. No gradients for the transform itself yet.

Disclaimer: First time TF contributor, feedback is welcome.

### CODE REVIEW

**Comment by vrv** [comment]
Thanks for the PR!

@tensorflow-jenkins test this please

@ringw can you please review for correctness and overall idea?

**Comment by vrv** [comment]
@tensorflow-jenkins test this please

**Review by ringw — APPROVED** [review]
Thanks, looks great!

**Comment by sebastian-schlecht** [comment]
@vrv Thanks for fixing pending linter issues 😊
